standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -4638,8 +4649,8 @@ public GetPartitionsResponse getPartitionsWithSpecs(GetPartitionsRequest request
   }
 
   @Override
-  public OptionalCompactionInfoStruct findNextCompact(String workerId) throws MetaException, TException {
-    return client.find_next_compact(workerId);
+  public OptionalCompactionInfoStruct findNextCompact(String workerId, String workerVersion) throws MetaException, TException {
```

Comment:
is this a backward compatible change?

Reply:
The thrift API changed, so old HS2 won't be able to connect to new HMS, but that is ok. The HMS backend db is backward compatible, only new fields were added, the old HMS versions will not use them

Reply:
I am not sure we could retire public API like this upstream. Not 100% sure, but I think we should keep prev method signature as well but mark it as deprecated.

Reply:
ok, added back the original method with deprecation.